packages/dd-trace/src/plugins/util/web.js

```diff
@@ -570,11 +572,16 @@ function getHooks (config) {
   return { request }
 }
 
-function getMiddlewareSetting (config) {
-  if (config && typeof config.middleware === 'boolean') {
-    return config.middleware
-  } else if (config && config.hasOwnProperty('middleware')) {
-    log.error('Expected `middleware` to be a boolean.')
+function getMiddlewareSetting (config, tracerConfig) {
+  if (config) {
+    if (typeof config.middleware === 'boolean') {
+      return config.middleware
+    } else if (config.hasOwnProperty('middlewareTracingEnabled')) {
```

Comment:
This should be done in the plugin manager, consistent with other similar options. While it could be argued that this is not ideal, changing how this works is out of scope of this PR.